File: backend/app/schemas/notification.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.enums import NotificationType


class NotificationRead(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    type: NotificationType
    title: str
    body: str
    link: str | None
    data: dict
    is_read: bool
    created_at: datetime


class UnreadCountRead(BaseModel):
    unread_count: int

File: frontend/src/types/notification.ts
export type NotificationType = "new_brand_media" | "influencer_post_published" | "slot_claimed" | "payment_confirmed";

export interface Notification {
  id: string;
  type: NotificationType;
  title: string;
  body: string;
  link: string | null;
  data: Record<string, unknown>;
  is_read: boolean;
  created_at: string;
}
